File: src/stackQueue/다리를지나는트럭.java
package stackQueue;

import java.util.*;

public class 다리를지나는트럭 {
    public int solution(int bridge_length, int weight, int[] truck_weights) {
        int answer = 0;
        Queue<Integer> q = new LinkedList<>();
        int allWeight = 0;
        int time = 0;

        for(int i : truck_weights){
            while(true){
\                if(q.isEmpty()){
                    q.offer(i);
                    allWeight += i;
                    time++;
                    break;
\                }else if(q.size() == bridge_length){
                    allWeight -= q.poll();
                }else
                if(weight >= allWeight + i){
                    q.offer(i);
                    allWeight += i;
                    time++;
                    break;
                }else{
                    q.offer(0);
                    time++;
                }
            }
        }
    }
    answer += time + bridge_length;

        return answer;
}
}

File: 스택+큐/다리를 지나는 트럭.py
def solution(bridge_length, weight, truck_weights):
    answer = 0
    bridge = [0 for i in range(bridge_length)]
    
    while len(truck_weights) > 0:
        answer += 1
        bridge.pop(0)
        
        if sum(bridge) + truck_weights[0] > weight:
            bridge.append(0)
            
        else:
            bridge.append(truck_weights[0])
            truck_weights.pop(0)
            
    answer += bridge_length
    
    return answer
